test/cluster/test_client_routes.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (C) 2025-present ScyllaDB
+#
+# SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+import asyncio
+import pytest
+import logging
+import time
+import uuid
+
+from test.cluster.conftest import skip_mode
+from test.pylib.manager_client import ManagerClient
+from test.pylib.rest_client import HTTPError
+from test.pylib.util import wait_for
+
+from cassandra.protocol import known_event_types
+
+logger = logging.getLogger(__name__)
+CLIENT_ROUTES_CHANGE_EVENT_NAME = "CLIENT_ROUTES_CHANGE"
+
+async def wait_for_expected_client_routes_size(cql, expected_routes_size):
+    async def expected_client_routes_size(cql, expected_size):
+        res = list(cql.execute("SELECT * FROM system.client_routes"))
+        logger.info(f"Got client routes, expected_size={expected_size}, res={res}")
+        if len(res) == expected_size:
+            return res
+        return None
+    await wait_for(lambda: expected_client_routes_size(cql, expected_routes_size), time.time() + 10)
```

Comment:
You can harden these tests a bit by waiting until all live nodes have the expected `client_routes` size. You can pass a list of hosts to `wait_for_expected_client_routes_size` and execute the statement for each host.